src/pubkey.h

```diff
@@ -200,9 +202,32 @@ struct CExtPubKey {
                a.chaincode == b.chaincode && a.pubkey == b.pubkey;
     }
 
-    void Encode(unsigned char code[74]) const;
-    void Decode(const unsigned char code[74]);
+    void Encode(unsigned char code[BIP32_EXTKEY_SIZE]) const;
+    void Decode(const unsigned char code[BIP32_EXTKEY_SIZE]);
     bool Derive(CExtPubKey& out, unsigned int nChild) const;
+
+    unsigned int GetSerializeSize(int nType, int nVersion) const
+    {
+        return BIP32_EXTKEY_SIZE+1; //add one byte for the size (compact int)
+    }
+    template <typename Stream>
+    void Serialize(Stream& s, int nType, int nVersion) const
+    {
+        unsigned int len = BIP32_EXTKEY_SIZE;
+        ::WriteCompactSize(s, len);
+        unsigned char code[BIP32_EXTKEY_SIZE];
+        Encode(code);
+        s.write((const char *)&code[0], len);
+    }
+    template <typename Stream>
+    void Unserialize(Stream& s, int nType, int nVersion)
+    {
+        unsigned int len = ::ReadCompactSize(s);
+        unsigned char code[BIP32_EXTKEY_SIZE];
+        assert(len == BIP32_EXTKEY_SIZE);
```

Comment:
IMO it'd be better to throw an exception (serialization framework seems to prefer std::ios_base::failure, see [here](https://github.com/bitcoin/bitcoin/blob/master/src/addrman.h#L343) and [here](https://github.com/bitcoin/bitcoin/blob/master/src/serialize.h#L287) )  which can at least be caught instead of killing the program immediately.

Of course we don't expect to deserialize these from the network, otherwise someone may crash the application, but it may be better to be consistent.